libnmstate/metadata.py

```diff
@@ -147,3 +153,25 @@ def _generate_link_master_metadata(ifaces_desired_state,
                         master_has_no_slaves_specified_in_desired):
                     set_metadata_func(
                         master_state, ifaces_desired_state[slave])
+
+
+def _generate_route_metadata(desired_state):
+    """
+    Save routes under interface IP protocol so that nm/ipv4.py or nm/ipv6.py
+    could include route configuration in `create_setting()`.
+    Currently route['next-hop-interface'] is mandatory.
+    """
+    for iface_name, routes in six.viewitems(desired_state.iface_routes):
+        if not iface_name:
+            continue
+        iface_state = desired_state.interfaces.get(iface_name, {})
```

Comment:
In the "merge" step the desired state has been already complemented with the interfaces of new routes.
Can we drop the question about the existence of the interface in the state?

Reply:
Done.

Reply:
I think @EdDev meant to not use `.get()` but `[iface_name]`.